stonesoup/feeder/base.py

```diff
@@ -1,14 +1,46 @@
 # -*- coding: utf-8 -*-
 """Base classes for Stone Soup feeder"""
+from abc import abstractmethod
+
 from ..base import Property
-from ..reader import DetectionReader
+from ..reader import Reader, DetectionReader, GroundTruthReader
+from ..buffered_generator import BufferedGenerator
 
 
-class Feeder(DetectionReader):
+class Feeder(Reader):
     """Feeder base class
 
+    Feeder consumes and outputs :class:`.State` data and can be used to
+    modify the sequence, duplicate or drop data.
+    """
+
+    reader = Property(Reader, doc="Source of detections")
```

Comment:
The base Feeder class previously inherited from DetectionReader but had a detector property, so I kept the same form.

Reply:
Is a `Feeder` _logically_ a `Reader`? Should they have the same interfaces? It seems not, given that the `Feeder` has a `reader` property.

If not (and given that `Reader` is an empty class definition, why not use:
`class Feeder(Base, BufferedGenerator):`. If it helps, I checked and the tests pass if you do this. I am most interested, though, in what is logical rather than than what works.

Reply:
It does seem rather odd. Like you said, it works perfectly fine without subclassing Reader.

Reply:
@sdhiscocks

Reply:
So the Feeder has the same interface as the readers (with the extra _reader_ property). This is similar to _Gater_ being a _Hypothesiser_. Components that take _Reader_ class, can take a _Feeder_ instead, where a feeder just wraps and modifies data said reader.

Reply:
OK. Happy this is resolved then,